src/tools/fuzzing.h

```diff
@@ -2636,6 +2667,154 @@ class TranslateToFuzzReader {
     return builder.makeMemoryFill(dest, value, size);
   }
 
+  Expression* makeTry(Type type) {
+    hangStack.push_back(nullptr);
+    auto* ret = builder.makeTry(makeMaybeBlock(type), makeCatch(type), type);
+    hangStack.pop_back();
+    return ret;
+  }
+
+  Expression* makeCatch(Type type) {
+    // exnref.pop is a pseudo instruction that should follow 'catch'
+    Expression* pop = builder.makePop(Type::exnref);
+
+    // With a small probability, just rethrow it and get done
+    if (oneIn(10)) {
+      return builder.makeRethrow(pop);
+    }
+
+    // Make a block. Note that we don't insert this break in breakableStack:
+    // Because 'exnref.pop' should follow 'catch' and we don't support blocks
+    // taking values yet, if there are instructions that break to this block,
+    // this block cannot be removed when writing binary, which will be a
+    // problem. We also don't make a label for this block.
+    auto* ret = builder.makeBlock();
+    ret->type = type;
+
+    // Now that we have to consume exnref.pop. We try one of these four things:
+    // 1. Drop it
+    // 2. Set to a local
+    // 3. Set to a global
+    // 4. Use it as an argument to a function
+    Expression* firstExpr = nullptr;
+    switch (upTo(4)) {
+      case 0:
+        firstExpr = builder.makeDrop(pop);
+        break;
+      case 1: {
+        auto& locals = typeLocals[Type::exnref];
+        if (locals.empty()) {
+          firstExpr = builder.makeDrop(pop);
+        } else {
+          firstExpr = builder.makeLocalSet(pick(locals), pop);
+        }
+        break;
+      }
+      case 2: {
+        auto& globals = globalsByType[Type::exnref];
+        if (globals.empty()) {
+          firstExpr = builder.makeDrop(pop);
+        } else {
+          firstExpr = builder.makeGlobalSet(pick(globals), pop);
+        }
+        break;
+      }
+      case 3: {
+        if (wasm.functions.empty()) {
+          firstExpr = builder.makeDrop(pop);
+        } else {
+          int tries = TRIES;
+          Function* target = nullptr;
+          while (tries-- > 0) {
+            target = pick(wasm.functions).get();
+            if (target->sig.results != Type::none ||
+                target->sig.params.size() == 0 ||
+                *target->sig.params.expand().begin() != Type::exnref) {
+              continue;
+            }
+            // we found a function that satisfies all conditions!
+            std::vector<Expression*> args;
+            args.push_back(pop);
+            for (unsigned i = 1; i < target->sig.params.size(); i++) {
+              args.push_back(make(target->sig.params.expand()[i]));
+            }
+            firstExpr = builder.makeCall(target->name, args, type, false);
```

Comment:
shouldn't there be a break after this line? seems like it will keep looping.

Reply:
OMG good catch. Fixed. I wonder if we have been generating calls at all with this bug.